docs/design/mscclpp-dsl-json-file.md

```diff
@@ -0,0 +1,478 @@
+  "protocol": "Simple",
+  "inplace": true,
+  "gpus": [
+    {
+      "id": 0,
+      "inputChunks": 1,
+      "outputChunks": 2,
+      "scratchChunks": 0,
+      "chunkGroups": 1,
+      "threadblocks": [
+        {
+          "id": 0,
+          "ops": [
+            {
+              "name": "rsignal",
+              "o_cids": [
+                {
+                  "id": 0
+                }
+              ],
+              "ctype": "memory",
+            },
+            {
+              "name": "rwait",
+              "i_cids": [
+                {
+                  "id": 0
+                }
+              ],
+              "ctype": "memory",
+            },
+            {
+              "name": "put",
+              "o_buff": [
+                {
+                "srcbuff": "o",
+                "srcoff": 0,
+                "dstbuff": "o",
+                "dstoff": 0,
+                "cnt": 1
+                }
+              ],
+              "o_cids": [
+                {
+                  "id": 0
+                }
+              ],
+              "ctype": "memory"
+            },
+            {
+              "name": "signal",
+              "o_cids": [
+                {
+                  "id": 0
+                }
+              ],
+              "ctype": "memory",
+            },
+            {
+              "name": "wait",
+              "i_cids": [
+                {
+                  "id": 0
+                }
+              ],
+              "ctype": "memory",
+            }
+          ],
+          "channels":[
+            {
+              "ctype": "memory",
+              "cids": [
+                0
+              ]
+            }
+          ],
+          "bufferCollection": [
+            {
+              "buffid": 0
+            }
+          ]
+        }
+      ],
+      "channels": [
+        {
+          "type": "memory",
+          "connectedTo": [
+            1
+          ]
+        }
+      ],
+      "bufferCollection": [
+        {
+          "rank": 1,
+          "buff": "i"
+        }
+      ]
+    },
+    {
+      "id": 0,
+      "inputChunks": 1,
+      "outputChunks": 2,
+      "scratchChunks": 0,
+      "chunkGroups": 1,
+      "threadblocks": [
+        {
+          "id": 0,
+          "ops": [
+            {
+              "name": "rsignal",
+              "o_cids": [
+                {
+                  "id": 0
+                }
+              ],
+              "ctype": "memory",
+            },
+            {
+              "name": "rwait",
+              "i_cids": [
+                {
+                  "id": 0
+                }
+              ],
+              "ctype": "memory",
+            },
+            {
+              "name": "put",
+              "o_buff": [
+                {
+                "srcbuff": "o",
+                "srcoff": 1,
+                "dstbuff": "o",
+                "dstoff": 1,
+                "cnt": 1
+                }
+              ],
```

Comment:
Same here. We can get the channel via channel id, but how to find the associate buffer. Need to refer buffer collection in the put operation?